backend/schedulerd/executor.go

```diff
@@ -301,3 +279,46 @@ func processCheck(ctx context.Context, executor Executor, check *types.CheckConf
 	}
 	return nil
 }
+
+func buildRequest(executor Executor, check *types.CheckConfig) (*types.CheckRequest, error) {
+	request := &types.CheckRequest{}
+	request.Config = check
+
+	ctx := types.SetContextFromResource(context.Background(), check)
+
+	// Guard against iterating over assets if there are no assets associated with
+	// the check in the first place.
+	if len(check.RuntimeAssets) != 0 {
+		// Explode assets; get assets & filter out those that are irrelevant
+		assets, err := executor.getStore().GetAssets(ctx)
```

Comment:
The only thing executor is used for here is getStore(). It would be better to pass the store into the function. Then, there would be no need to include getStore() in the methods of executor.